### PR TITLE
Reward programs guides

### LINKED ISSUE #550: [Guide]Supernodes
Following the discussion in https://nemgroup-workspace.slack.com/archives/C015AEQ9ZL2/p1608557174425400.
New Concept page describing:
- What is the supernode program
- Where the min specs can be found

New Guide explaining:
- How to setup a supernode, including security (reference #549).
- Deploy the agent
- Send enrolment
- Meet various test
- Everything can be done with `symbol-bootstrap` (Described in nemtech/symbol-bootstrap#92)
- Take into account https://github.com/nemtech/symbol-bootstrap/pull/92#issuecomment-760903318

### PR DESCRIPTION
**Pull request by segfaultxavi**

### CODE REVIEW

File: source/concepts/rewards.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+============
+
+Supernodes are regular nodes meeting the following additional hardware and operational requirements:
+
+.. csv-table::
+    :header: "Parameter", "Requirement"
+    :delim: ;
+    :widths: 30 70
+
+    **Node type**; Must be a Dual node (Both a :ref:`Peer <peer-node>` and an :ref:`API <api-node>` node).
```
Comment (fboucquez) [inline source/concepts/rewards.rst:29]: A supernode must also include a rest gateway

Comment (segfaultxavi) [reply source/concepts/rewards.rst:29]: How does one include that? I was under the impression API nodes already provided REST access... What is exactly a REST gateway **from a user's perspective**?

File: source/guides/network/supernode-enrollment.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+##################################
+Enrolling to the Supernode program
+##################################
+
+**********************
+Using symbol-bootstrap
+**********************
```
Comment (fboucquez) [inline source/guides/network/supernode-enrollment.rst:7]: You can base this guide on this procedure 

https://github.com/nemtech/symbol-bootstrap/pull/92

File: source/guides/network/supernode-enrollment.rst [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+********
+Manually
+********
+
+- Enrols as a supernode (msg on chain), content ``enrol ${nodePublicKey} ${agentUrl}``
+
+  - supernodeEnrolmentAddress: TDETDMT5S2ADAYGJXPATUHQUYVGSLSVJ6TLSXQQ for TESTNET
+
+- Installs an agent to monitor it. source https://github.com/nemgrouplimited/symbol-node-rewards, binary in S3, url??? platforms???
```
Comment (fboucquez) [inline source/guides/network/supernode-enrollment.rst:17]: The agent binary can be publically downloaded from 

https://symbol-node-reward.s3-eu-west-1.amazonaws.com/agent_binary/agent-linux.bin

Atm, we are just deploying the Linux version of the binary, we would need to upgrade adding other distributions. Unsure if they are going to be windows or mac supernodes. We would need to test the agents on the different distributions. 

Note, bootstrap pulls (and caches) this Linux binary from the s3 location when setting up the node. When using --upgrade, the latest binary will be pulled as part of the node upgrade.

File: source/guides/network/supernode-enrollment.rst (hunk 2 above)
Comment (segfaultxavi) [reply source/guides/network/supernode-enrollment.rst:7]: Yeah, that's the plan!

**Comment by segfaultxavi** [comment]
Thanks @fboucquez for the review, but this is still a draft PR, bound to be heavily edited :)

File: source/concepts/rewards.rst (hunk 1 above)
Comment (fboucquez) [reply source/concepts/rewards.rst:29]: "The node must include a Rest Gateway"

The "full" dual node has 4 components:
- Server node in peer and api mode. Port 7900
- Broker that keeps the mongo DB up to date and provides the WebSockets queues. Port 7902
- Mongo db
- Rest Gateway (aka catapult-rest) that reads data from the mongo db. Port 3000. This the facade for regular apps like explorer, cli, wallet, etc.

Our api/dual presets on bootstrap add rest gateways by default. But if you are manually creating a supernode, you need to install the javascript catapult-rest.

The theory goes, the user could have a dual node without a rest gateway. Supernode program requires a rest gateway so services like wallet, explore could potentially talk to the node. The node monitor also uses it to validate the state of the node (like height and server version).

Comment (segfaultxavi) [reply source/concepts/rewards.rst]: I've added a note, and will include the information regarding the REST Gateway once I write the guide on setting up a node that includes a REST Gateway :)

**Comment by mm-s** [comment]
Under section `Super-node program.requirements` `Bonded Deposit` defines almost imperceptibly the word `tier`.
The next section, `Rewards`, explains them breaking them down in 'tiers', assuming a knowledge of what tier means.

I suggest a hyperlink to its definition for those stepping directly into the rewards section.

**Comment by segfaultxavi** [comment]
I do not understand what is the part that is obscure. First it talks about "tiers" and says (see below), so you know that they will be defined later. Then tiers are defined as "Supernodes are divided in 3 tiers according to their staked symbol.xym: More than 1 million, more than 2 million and more than 3 million."

What exactly would you change?

**Comment by segfaultxavi** [comment]
Closing this PR. It has been merged with #581 into #586. Both PR were related and it was hard to review them separately.

**Pull request closed without merging**